### DISCUSSION TITLE
Client keyboard mouse/keyboard  no work

### DISCUSSION THREAD

**Discussion by brcisna** [ltsp/ltsp#272] [Q&A]
Hello All,

Client - Dell All-In-One 4 years old
Debian Bullseye (11)

After updating the server , along with the client image and rebooting both, the client keyboard & mouse no longer works. Have the login screen showing but neither will function. tried two different keyboards, mice. Am using lightdm display manger which have used for about 6 months now.
The server keyboard ,mouse, works as expected.
This was an update from kernel 5.4.x to 5.5.x .

Thank You

**Comment by alkisg** [comment]
Hi, this usually indicates a mismatched kernel, where the modules at `/lib/modules/$(uname -r)` aren't there because the older kernel was removed as is only there in TFTP.

For example, if you're using a chroot and you update it, then you need to run `ltsp kernel chroot` for `/srv/tftp/ltsp/chroot/vmlinuz` to match `/srv/ltsp/chroot/lib/modules/$(uname -r)`.

**Comment by brcisna** [comment]
Hi Alkisg,

OK I am showing my stupidity here.

so i run 
1. ltsp kernel chroot /srv/tftp/ltsp/chroot/vmliniz
2. ltsp kernel chroot /srv/ltsp/chroot/lib/modules/$(uname -r)
3. reboot client

and should be golden then?

Also i have no chroot directory showing at /srv/tftp/ltsp/ ? This is a default ltsp installation about 6 months old now.

& yes,,,with this kernel update i had to do an apt autoremove which removed an older kernel as i was out of space on the boot partition

When I run:  sudo ltsp kernel chroot
Image does not exist: chroot
Image doesn't exist: 
LTSP command failed: test kernel_main: != kernel_main:
Aborting ltsp



Thanks.

**Comment by alkisg** [comment]
@brcisna, what's the output of `ltsp info`?

When I was saying "chroot" above, it was an example, to replace it with your actual chroot name. Do you have a chroot? Are you using chrootless? `ltsp info` will tell us.

**Comment by brcisna** [comment]
Alkisg,

LTSP version: 20.04-1

SERVER XSESSIONS: gnome-classic.desktop  lightdm-xsession.desktop  remmina-gnome.desktop
gnome.desktop	       LXDE.desktop
gnome-xorg.desktop     openbox.desktop

SERVER OS: PRETTY_NAME="Debian GNU/Linux bullseye/sid"

CHROOTS:


VMs:


IMAGES:
x86_64
x86_64_nokeyboardmouse

'nokeyboardmouse' i named to get back to working image ,FYI.

Also,, as an aside,, I notice the generated client image is many megabytes smaller than good working image?

Thank You
Barry

**Comment by alkisg** [comment]
Barry do you have epoptes installed?

If so, boot with the broken image, right click on a client, select "open terminal > root > locally", and run the following commands, and paste the output here:

```shell
uname -a
dpkg -l '*linux*' | grep ^ii
```

If you don't have epoptes, you might be able to press Ctrl+Alt+F2 and login to vt2 and run the commands directly on the client.

**Comment by brcisna** [comment]
Alkis,,

Thank You,,, 

but if i try and boot the client with broken image i get no keyboard and mouse to be able to type in anything?
I am not explaining myself clear enough i think?

You are saying if i can remote into server,,, client  terminal these commands you have sent me?

**Comment by alkisg** [comment]
Yes, epoptes is remote control, so you use the server keyboard.

Also, Ctrl+Alt+F2 with the client keyboard might work even if typing in lightdm doesn't work.

**Comment by brcisna** [comment]
uname - a
Linux server1 5.5.0-2-amd64 ltsp/ltsp#161 SMP Debian 5.5.17-1 (2020-04-15) x86_64 GNU/Linux

dpkg -l '*linux*' | grep ^ii
ii  binutils-x86-64-linux-gnu          2.34-5       amd64        GNU binary utilities, for x86-64-linux-gnu target
ii  console-setup-linux                1.195        all          Linux specific part of console-setup
ii  firmware-linux-free                20200122-1   all          Binary firmware for various drivers in the Linux kernel
ii  firmware-linux-nonfree             20190717-2   all          Binary firmware for various drivers in the Linux kernel (meta-package)
ii  fonts-linuxlibertine               5.3.0-4      all          Linux Libertine family of fonts
ii  libselinux1:amd64                  3.0-1+b3     amd64        SELinux runtime shared libraries
ii  libselinux1-dev:amd64              3.0-1+b3     amd64        SELinux development headers
ii  linux-base                         4.6          all          Linux image base package
ii  linux-compiler-gcc-9-x86           5.5.17-1     amd64        Compiler for Linux on x86 (meta-package)
ii  linux-headers-5.5.0-1-amd64        5.5.13-2     amd64        Header files for Linux 5.5.0-1-amd64
ii  linux-headers-5.5.0-1-common       5.5.13-2     all          Common header files for Linux 5.5.0-1
ii  linux-headers-5.5.0-2-amd64        5.5.17-1     amd64        Header files for Linux 5.5.0-2-amd64
ii  linux-headers-5.5.0-2-common       5.5.17-1     all          Common header files for Linux 5.5.0-2
ii  linux-headers-amd64                5.5.17-1     amd64        Header files for Linux amd64 configuration (meta-package)
ii  linux-image-5.5.0-1-amd64          5.5.13-2     amd64        Linux 5.5 for 64-bit PCs (signed)
ii  linux-image-5.5.0-2-amd64          5.5.17-1     amd64        Linux 5.5 for 64-bit PCs (signed)
ii  linux-image-amd64                  5.5.17-1     amd64        Linux for 64-bit PCs (meta-package)
ii  linux-kbuild-5.5                   5.5.17-1     amd64        Kbuild infrastructure for Linux 5.5
ii  linux-libc-dev:amd64               5.5.17-1     amd64        Linux support headers for userspace development
ii  util-linux                         2.34-0.1     amd64        miscellaneous system utilities
ii  util-linux-locales                 2.34-0.1     all          locales files for util-linux

Thank You.

**Comment by alkisg** [comment]
Are these from the broken client, or from the server?

Barry do you want to come on IRC, for remote support? It should be a lot faster...

Go to ltsp.org and click on IRC.

**Comment by brcisna** [comment]
Alkisg,

Also looking in my apt log for removed things i see this,,,:

tart-Date: 2020-04-25  10:11:37
Commandline: apt upgrade
Requested-By: brcisna (1001)
Upgrade: pkg-config:amd64 (0.29-6, 0.29.2-1)
Error: Sub-process /usr/bin/dpkg returned an error code (1)
End-Date: 2020-04-25  10:12:14

This may be from boot directory out of space though,,,which has been updated since.

**Comment by brcisna** [comment]
OK going there,,,thanks

**Comment by alkisg** [comment]
If you ran `ltsp image /` while apt was in broken state, then that might be the problem.

**Comment by alkisg** [comment]
Barry as we said I rebooted the client, but somehow the server was disconnected from me. I'll be online for a few more minutes, if you can't make it we can continue some other day.

**Comment by brcisna** [comment]
Alkisg,

Still no keyboard mouse,,,, i renamed OLD image to get a working clinet,
I certainly appreciate your efforts,,
My main concern is now the client image is almost 700 mb smaller than good image... i didnt delete any packages in the interim of these two,,other than kernel update?


Dont know what else to try at this point?

Thanks again<
Barry

**Comment by brcisna** [comment]
alkisg,

Just wanted to add. I see all kinds of spam stuff in logs from 'tracker-' i have never seen this before. It looks like some sort of metadata storage framework. Is there any chance this may be causing this problem? I am sure i never installed it. I do install all kinds of multimedia  packages as i do old history videos in a  novice way.

Thank You for all the time you spent troubleshooing.

Barry

**Comment by alkisg** [comment]
Barry since everything is working fine over epoptes, it's hard to troubleshoot remotely.

I feel it's not LTSP related at all, but it's due to Bulleye not being stable yet. Maybe even related to the fact that you installed a whole lot of overlapping desktop environments and display managers. You might want to use Buster, or contact the Debian community for this...

**Comment by brcisna** [comment]
Yes,

Thank you for all the time you spent on this troubleshooting.
I'll do a new install of Buster and get back to square one.
I have for example about two months ago installed virtualenv to get a
couple of standalone multimedia programs to work,and just a lot of clutter
as you said.
When i get to work tomorrow i am going to get a ps2 mouse and keyboard just
to see if they work on this Dell All-in-one. I noticed this does in fact
have a ps2 keyboard and mouse port on it.
Used to keep all that extra stuff here at house and got tired of seeing
that old stuff setting around.

Edit: Alkis,
Wanted to add just to try and learn something with the latest/broken image,  i  booted a pxeclient in virtualbox residing at the server. I noticed after getting to the login screen the nic activity indicator is very active for almost 5 minutes then stops,almost as though it is in a loop or something. I always set this up for troubleshooting after i get ltsp installed on the server. Have used ltsp since it's inception almost 20 years now.

Thanks again,

Barry

**Comment by brcisna** [comment]
Alkis,

Not to keep harping on this,,,but i found on client at boot startup if i try and do a ctl-b to get into ipxe config my keyboard is not seen, nothing happens . It is the same at the blue menu,,,i cannot scroll up or down to select anything other the default x86_64.img.
Not sure if this is expected behavior? I
 guess i have never tried to use this until now,,just trying to troubleshoot.

Thanks again

**Comment by alkisg** [comment]
Barry, if you're using a separate boot partition, you need to use the following command in order to include the /boot directory in the LTSP image:

```shell
ltsp image /,,/boot,subdir=boot
```

Documentation about it is in `man ltsp ipxe`, section "advanced image sources".

**Comment by brcisna** [comment]
Hi Alkisg,

ltsp image /,,/boot,subdir=boot

This command made the client working again!
Thank You so much & also to vagrantc

Thinking back on this i think you may have given me this command when i first switched over to new ltsp  and couldnt get a booting client, but i totally forgot about it,I am guessing was always using the original /boot directory with updated binaries on the client.  Have always just used the default ltsp image command . Maybe moving to the newer 5.5.x kernel was enough to not let things work without updated /boot dir.


I need to read up more on the documentation for each of the ltsp commands.
I would have never figured out this command without you mentioning it, anyway. 
I about drove you and vagrant both crazy probably.

Thanks again

**Comment by alkisg** [comment]
I think the most misleading thing in this case, is that the underlying file system under /boot already had a matching but broken kernel, maybe due to your troubleshooting of the /boot partition free space.

Normally, this use case would stop with "could not locate vmlinuz", so we'd immediately pinpoint it.

**Comment by brcisna** [comment]
Makes sense Alkis,

 About 4 days ago ,,This is probably where when doing the TWO kernel updates 5.5-1 & 5.5-2 i apt was running , along with a few other updates it failed to complete,with "/boot drive is full."  & failed to complete the initrd routine. Then the suggested "'run autoremove to free space". by removing two older kernel,s then apt run the initrd routine. That wording isn't  completely correct but close.

You could mark this issue as closed.

Thank You
